packages/components/src/styles/components/form/layout.scss

```diff
@@ -0,0 +1,76 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+//
+// FORM LAYOUT COMPONENTS
+//
+
+@use "../../mixins/breakpoints" as *;
+
+// FORM
+
+.hds-form {
+  --hds-form-section-max-width: 700px; // default max-width used by FormSection, etc.
+
+  display: flex;
+  flex-direction: column;
+  gap: 32px;
+}
+
+// FORM HEADER & SECTION HEADER
+
+.hds-form__header,
+.hds-form__section-header {
+  display: flex;
+  flex-direction: column;
+  gap: 8px;
+}
+
+// FORM SECTION
+
+.hds-form__section {
+  display: flex;
+  flex-direction: column;
+  gap: 24px;
+}
+
+// hasBorder = true
+.hds-form__section--has-border {
+  padding: 24px;
+  border: 1px solid var(--token-color-border-primary);
+  border-radius: var(--token-border-radius-medium);
+}
+
+// FIELD GROUP
+// Note: Uses a Layout::Flex component under the hood which includes the main layout styles
+// If responsive behavior is later added to the Layout::Flex component, this behavior should be replaced and
+// the styles should be removed
+
+.hds-form__section-field-group {
+  &.hds-layout-flex--direction-row {
+    // When there are columns, stack them vertically
+    @include hds-breakpoint-below("md") {
```

Comment:
[issue] there is an issue with applying a `basis` value to the items, when this becomes a stacked layout: 
<img width="1537" alt="screenshot_5066" src="https://github.com/user-attachments/assets/f8131547-e98a-49e3-84b0-ffa2abe3cddb" />

Reply:
I don't know if this is a problem. I'm seeing "basis" here more as the "width" that you want the field to be which would be the same regardless of whether it stacks.

I don't see why you'd want a field for a zip code, for example, to expand to full width just because the layout "stacks" otherwise you'd just set it to be full-width for the wider view too in my mind.

Reply:
So it seems it may make more sense to instead add a `width` option to the `SectionFieldGroup`. And if we did that, the `Item` wouldn't really be needed. So maybe in that case it would actually make sense to do a custom implementation for the layout vs. trying to reuse the `Layout::Flex` component after all...

Reply:
Mmm how would the width cascade down to the items? If, as a consumer, I want to have different widths for different "group items", how could I achieve that?

Reply:
One thing I was thinking, is that we may override* the `basis` with the `initial` value (or another value that makes sense) when it's stacked, because in that case everything just needs to have its own intrinsic height.

Reply:
Oh, you're right. I was actually thinking of the individual Input items inside needing a width. (I confused myself...)

Really, I think the `Item` could use a width property...

Reply:
What do you think about making an `SectionFieldGroup::Item` and adding a `@width` property to it?